extra/intl-extra/IntlExtension.php

```diff
@@ -372,7 +372,7 @@ public function formatDateTime(Environment $env, $date, ?string $dateFormat = 'm
         $date = CoreExtension::dateConverter($env, $date, $timezone);
 
         $formatterTimezone = $timezone;
-        if (null === $formatterTimezone) {
+        if (null === $formatterTimezone || false === $formatterTimezone) {
```

Comment:
This seems to create a different behaviour than stated in the docblock

```php
    /**
     * @param \DateTimeInterface|string|null  $date     A date or null to use the current time
     * @param \DateTimeZone|string|false|null $timezone The target timezone, null to use the default, false to leave unchanged
*/
```

Reply:
Can you give more details?

In the docblock:
> false to leave unchanged

And as you can see in the test case, the original UTC date time is 1pm is still 1pm even if twig has been configured on another timezone.

BTW, I did not change the logic, just fixed the regression introduced by https://github.com/twigphp/Twig/pull/3903

Reply:
You're right, but i think "null" and "false" should not be handled in the same way ... at least according to the documentation

> null to use the default, false to leave unchanged

And i'm not saying this is your fault :)

Reply:
Yes, this is handled differently inside `CoreExtension::dateConverter()` just before the `if`

Reply:
Damn.. sorry, it took me times to understand the "reversion" :(